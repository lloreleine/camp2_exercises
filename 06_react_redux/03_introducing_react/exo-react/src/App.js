import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import _ from 'underscore';

function Row(props){
  return (
    <tr>
      <td>{props.arrayProducts.decathlon_id}</td>
      <td>{props.arrayProducts.title}</td>
      <td>{props.arrayProducts.price}</td>
    </tr>
  )
}

class App extends Component {
  constructor(props){
    super(props)
    this.state = {
      allProducts: props.products,
      order: {decathlon_id:"asc",title:"asc",price:"asc"}
    }
  }

  sortby = (filter) => {
    console.log(this.state.order);
    console.log(this.state.order[filter]);
    let newOrder = this.state.order;
    if(this.state.order[filter]==="asc"){
      let sortedproducts = _.sortBy(this.state.allProducts, filter).reverse()
      newOrder[filter] = "desc";
      this.setState({ allProducts: sortedproducts, order:newOrder })
    } else {
      let sortedproducts = _.sortBy(this.state.allProducts, filter)
      newOrder[filter] = "asc";
      this.setState({ allProducts: sortedproducts, order:newOrder  })
    }
  }

  render() {
    let sortedProducts = this.state.allProducts;
    return (
      <table>
        <thead>
          <tr>
            <th onClick={() => this.sortby('decathlon_id')}>Decathlon Id</th>
            <th onClick={() => this.sortby('title')}>Title</th>
            <th onClick={() => this.sortby('price')}>Price</th>
          </tr>
        </thead>
        <tbody>
          {sortedProducts.map(prdct => <Row key={prdct.decathlon_id} arrayProducts={prdct}/>)}
        </tbody>
      </table>
    );
  }
}


export default App;
