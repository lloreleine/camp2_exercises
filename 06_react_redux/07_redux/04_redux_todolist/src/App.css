.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

.margin {
  margin:10px;
}

.strikethrough {
  text-decoration: line-through;
  color: red;
}

.counter {
  margin-left: 20px;
}

.buttonsFilters {
  margin-top: 20px;
  margin-left: 5px;
  margin-right: 10px;
  border-color: skyblue;
  background-color: #fff;
}


@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
