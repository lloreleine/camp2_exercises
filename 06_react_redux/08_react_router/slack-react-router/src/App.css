.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

.margin-top {
  margin-top: 10px;
}

.red{
  color:red;
  font-weight: bold;
  padding: 5px 10px;
}

.blue{
  color:blue;
  font-weight: bold;
  padding: 5px 10px;
}

.bloc-channels{
  display: inline-block;
  margin-top:10px;
  margin-right: 10px;
}

.div_scroll{
  display: block;
  height: 100px;
  overflow: scroll;
}

.channel-container h3, .channel-container h4{
  margin:5px;
  padding:5px;
}

.channel-container h4{
  font-weight: lighter;
  font-style: italic;
  font-size: 12px;
}

.channel-container h5{
  font-weight: normal;
  text-decoration: underline;
}

.channelPurple{
  color: #660066;
}

.channelBlue{
  color: #468499;
}

.message-container {
  margin-top:10px;
  width: 100%;
  text-align: left;
  display: block;
  height: 150px;
  overflow: scroll;
}

.chatSendMsg{
  height: 50px;
  display: flex;
  padding-left: 10px;
  padding-right: 10px;
}

.chatSendMsg form {
  display: flex;
  width: 100%;
}

.chatSendMsg input {
  flex: 1 0;
}

.chatSendMsg button {
  height: 100%;
  margin: 0;
  padding-left: 20px;
  padding-right: 20px;
  background-color: #222;
  color: white;
  border: none;
  -webkit-appearance: none;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
