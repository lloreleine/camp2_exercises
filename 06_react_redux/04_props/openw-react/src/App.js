import React, { Component } from 'react';
import logo from './logo.svg';
import CallApi from './CallApi';
import './App.css';

const API_KEY = process.env.REACT_APP_OPENWEATHER_API_KEY2;

function fetchWeather(city){
  return fetch(`http://api.openweathermap.org/data/2.5/weather?q=${city}&units=metric&APPID=${API_KEY}`)
    .then(response => response.json())
    .then(function(result){
      let arrayWeather = [];
      arrayWeather.push(result.main.temp);
      arrayWeather.push(result.weather[0].main);
      return arrayWeather;
    })
}


class App extends Component {
  constructor(props){
    super(props);
    this.state = {
      city: '',
      weather: [],
      temp : '',
      loading: false,
      error: false
    };
  }

  handleInput = (evt) => {
    this.setState({city: evt.target.value});
  }

  submitOpenW = (evt) => {
    evt.preventDefault();
    this.setState({loading: true}, () =>
      fetchWeather(this.state.city)
      .then(temp => {
        this.setState({
          weather: temp,
          loading: false
        })
      })
    )
  }

  displayWeather() {
    if(this.state.weather[0]){
      return (
        <p>Weather: {this.state.weather[1]}<br/>
        Temperature: {this.state.weather[0]}Â°C
        </p>
      )
    }
  }
  render() {
    const loadPage = <h5>Loading...</h5>
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Open Weather</h1>
        </header>
        <p className="App-intro">
        </p>
        <CallApi state={this.state} handleInput={this.handleInput} submitOpenW={this.submitOpenW}/>
        <div>
        {this.state.loading && loadPage}
        {this.displayWeather()}
        </div>
      </div>
    );
  }
}

export default App;
export { fetchWeather };
