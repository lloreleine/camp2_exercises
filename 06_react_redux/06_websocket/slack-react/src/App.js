import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import Form from './Form';
import Chat from './Chat';
import Message from './Message';

class App extends Component {
  constructor(props){
    super(props);
    this.handleInput = this.handleInput.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.submitMsg = this.submitMsg.bind(this);
    this.handleMsg = this.handleMsg.bind(this);
    this.ws = new WebSocket("ws://localhost:4000");

    this.state = {
      current: '',
      isLoggedIn: false,
      nbUsers:0,
      usersList: [],
      currentMsg: '',
      msgsList:[]
    };
  }

  componentDidMount(){
    this.ws.onmessage = (event) => {
      let newMsgsList = this.state.msgsList;
      newMsgsList.push(JSON.parse(event.data));
      this.setState({
        msgsList: newMsgsList,
        nbUsers:event.data.nbUsers
      })
    };

    // Alert the server that the client is gone
    window.addEventListener("beforeunload", () => this.ws.send("CLOSE"));
  }

  handleInput(event) {
    this.setState({
      current: event.target.value
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    let newListUser = this.state.usersList;
    newListUser.push({name: this.state.current, isLogged: true});

    this.setState({
      usersList: newListUser,
      isLoggedIn: true
    });
  }

  handleMsg(event) {
    this.setState({
      currentMsg: event.target.value
    });
  }

  submitMsg(event) {
    event.preventDefault();
    let newMsg = {
      nameUser: this.state.current,
      text: this.state.currentMsg
    };
    // newListMsgs.push({nameUser: this.state.current, text: this.state.currentMsg});
    this.ws.send(JSON.stringify(newMsg));
  }

  render() {

    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to Slacky React</h1>
        </header>
        <div className="margin-top">
        {this.state.isLoggedIn === false &&
          <Form user={this.state.user} handleInput={this.handleInput} handleSubmit={this.handleSubmit} />
        }
        {this.state.isLoggedIn &&
          <Chat userLoggedIn={this.state.usersList[0]} submitMsg={this.submitMsg} handleMsg={this.handleMsg}/>
        }
        </div>

        {this.state.isLoggedIn === true &&
        <div>
          <h4>Slacky Chat</h4>
          <table>
            <thead>
              <tr>
                <th>From</th>
                <th>Message</th>
              </tr>
            </thead>
            <tbody>
              {this.state.msgsList.map((msg,i) => <Message key={i} msg={msg} name={this.state.current} />)}
            </tbody>
          </table>
        </div>
        }

      </div>
    );
  }
}

export default App;
