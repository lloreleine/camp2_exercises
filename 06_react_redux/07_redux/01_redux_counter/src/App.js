import React from 'react';
import logo from './logo.svg';
import './App.css';
import { connect } from 'react-redux';
import store from './Store';
import ConnectedTitle from './Title';


function App(props) {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <h1 className="App-title">Counter React - Redux</h1>
      </header>
      <p className="App-intro">
        <ConnectedTitle />
        <button onClick={props.decrement}>-</button>
        {props.counter}
        <button onClick={props.increment}>+</button>
      </p>
    </div>
  );
}


function mapCounterToProps(state) {
  return {
    counter: state.counterCombine.counter,
  }
}

function mapDispatchToProps(dispatch) {
  return {
    increment: () => dispatch({type: "INCREMENT"}),
    decrement: () => dispatch({type: "DECREMENT"})
  }
}


const ConnectedApp = connect(mapCounterToProps, mapDispatchToProps)(App);
export default ConnectedApp;
