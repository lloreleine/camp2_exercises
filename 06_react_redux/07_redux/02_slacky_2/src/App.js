import React, { Component } from "react";
import { connect } from 'react-redux';
import "./App.css";
import ConnectedLogin from "./Login";
import ConnectedChat from "./Chat";

const websocket = new WebSocket("ws://localhost:3000");

websocket.addEventListener("message", event => {
    const message = JSON.parse(event.data);
    console.log("Message from server ", message);
    switch (message.type) {
      case "CONNECTION_START":
      default:
        return;
      case "MESSAGES":
        return;
    }
  })

function App(props) {

  // componentDidMount() {
  //   // Listen for messages
  //   this.websocket.addEventListener("message", event => {
  //     const message = JSON.parse(event.data);
  //     console.log("Message from server ", message);
  //     switch (message.type) {
  //       case "CONNECTION_START":
  //       default:
  //         return;
  //       case "MESSAGES":
  //         this.setState({ messages: message.data });
  //         return;
  //     }
  //   });
  // }
  //
  // handleUserName = userName => {
  //   this.setState({ userName: userName });
  //   this.websocket.send(
  //     JSON.stringify({
  //       type: "LOGIN",
  //       userName: userName
  //     })
  //   );
  // };
  //
  // sendMessage = message => {
  //   this.websocket.send(
  //     JSON.stringify({
  //       type: "NEW_MESSAGE",
  //       userName: this.state.userName,
  //       message: message
  //     })
  //   );
  // };

  return (
    <div className="App">
      <header className="App-header">
        <h1 className="App-title">Slacky</h1>
      </header>

      {props.userName ? (
        <ConnectedChat sendMessage={props.sendMessage} messages={props.messages} />
      ) : (
        <ConnectedLogin handleChange={props.handleChange} />
      )}
    </div>
  );
}

function mapAppToProps(state){
  return {
    currentValue: state.handleChange.currentValue,
    userName: state.handleSubmit.userName
  }
}

function mapDispatchToProps(dispatch){
  return {
    handleChange: () => dispatch({type: "LOGIN_USER"})
  }
}

const ConnectedApp = connect(mapAppToProps, mapDispatchToProps)(App);
export default ConnectedApp;
