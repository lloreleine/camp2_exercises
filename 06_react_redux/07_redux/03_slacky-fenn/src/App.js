import React, { Component } from "react";
import "./App.css";
import Login from "./Login";
import Chat from "./Chat";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userName: null,
      messages: [],
    };
    // Attaching the websocket to our App so we can reuse it
    this.websocket = new WebSocket(`ws://${window.location.hostname}:8080`);
  }
  componentDidMount() {
    // Listen for messages
    this.websocket.addEventListener("message", event => {
      const message = JSON.parse(event.data);
      console.log("Message from server ", message);
      switch (message.type) {
        case "CONNECTION_START":
        default:
          break;
        case "MESSAGES":
          this.setState({ messages: message.data });
          break;
      }
    });
  }

  handleUserName = userName => {
    this.setState({ userName: userName });
    this.websocket.send(
      JSON.stringify({
        type: "LOGIN",
        userName: userName
      })
    );
  };

  sendMessage = message => {
    this.websocket.send(
      JSON.stringify({
        type: "NEW_MESSAGE",
        userName: this.state.userName,
        message: message
      })
    );
  };

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Slacky</h1>
        </header>

        {this.state.userName
          ? (<Chat sendMessage={this.sendMessage} messages={this.state.messages} />
          )
          : (<Login handleUserName={this.handleUserName} />)}
      </div>
    );
  }
}

export default App;
