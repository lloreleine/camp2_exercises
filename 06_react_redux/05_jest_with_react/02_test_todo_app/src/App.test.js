import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

beforeAll(() => {
  configure({ adapter: new Adapter() });
});

test('Everything looks the same', () => {
  const component = renderer.create(<App />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("SetState with a new task when enter something", () => {
  const testInput = 'Do the laundry';
  const load = shallow(<App />);
  load.find("form input").simulate("change", {target: {value: testInput}});
  // console.log(load.state())
  expect(load.find("form input").props().value).toEqual(testInput);
});

test("Display a new task when click on Add", () => {
  const testInput = 'Do the laundry';
  const load = shallow(<App />);
  load.find("form input").simulate("change", {target: {value: testInput}});
  // [ { id: 1, label: 'Do the laundry', completed: true } ]

  load.find("form").simulate("submit", { preventDefault: () => {} });
  expect(load.find("li span").text()).toBe(testInput);
});

test("Add a delete button when a task is checked", () => {
  const testInput = 'Do the laundry';
  const load = shallow(<App />);
  load.find("form input").simulate("change", {target: {value: testInput}});
  load.find("form").simulate("submit", { preventDefault: () => {} });

  let testState = load.state();
  load.find("li input").simulate("change", 2);
  // [ { id: 2, label: 'Do the laundry', completed: true } ]

  expect(testState.tasks[0]).toMatchObject({"label": testInput});
});

test("Add 2 tasks and store those two in tasks:[] when checked", () => {
  expect.assertions(4);
  const testInput1 = 'Buy some bread';
  const testInput2= 'Wish a happy birthday to Medhi';
  const load = shallow(<App />);
  load.find("form input").simulate("change", {target: {value: testInput1}});
  load.find("form").simulate("submit", { preventDefault: () => {} });
  expect(load.find("li").at(0).find("span").text()).toBe(testInput1);

  load.find("form input").simulate("change", {target: {value: testInput2}});
  load.find("form").simulate("submit", { preventDefault: () => {} });
  expect(load.find("li").at(1).find("span").text()).toBe(testInput2);

  load.find("li").at(0).find("input").simulate("change", 3);
  load.find("li").at(1).find("input").simulate("change", 4);
  // [ { id: 3, label: 'Buy some bread', completed: true } ]
  // [ { id: 4, label: 'Wish a happy birthday to Medhi', completed: true } ]

  expect(load.state().tasks[0]).toMatchObject({"label": testInput1});
  expect(load.state().tasks[1]).toMatchObject({"label": testInput2});
});

// test('Array with Object Containing', () => {
//   expect(
//     [
//       { a: 1, b: 2 },
//       { c: 1, d: 1 }
//     ])
//     .toContainEqual(
//       expect.objectContaining({ a: 1, b: expect.anything() })
//     )
// })

test("Delete a task", () => {
  expect.assertions(2);
  const testInput1 = 'Buy some bread';
  const testInput2= 'Wish a happy birthday to Medhi';
  const load = shallow(<App />);
  load.find("form input").simulate("change", {target: {value: testInput1}});
  load.find("form").simulate("submit", { preventDefault: () => {} });
  load.find("form input").simulate("change", {target: {value: testInput2}});
  load.find("form").simulate("submit", { preventDefault: () => {} });

  load.find("li").at(0).find("input").simulate("change", 5);
  load.find("li").at(1).find("input").simulate("change", 6);
  // [ { id: 5, label: 'Buy some bread', completed: true } ]
  // [ { id: 6, label: 'Wish a happy birthday to Medhi', completed: true } ]

  load.find("li input").at(3).simulate("click", 6);

  expect(load.state().tasks[0]).toMatchObject({"label": testInput1});
  expect(load.state().tasks.length).toEqual(1);
});
