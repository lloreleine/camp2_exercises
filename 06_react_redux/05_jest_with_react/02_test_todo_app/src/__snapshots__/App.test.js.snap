// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Everything looks the same 1`] = `
<div
  className="Todo"
>
  <div
    className="new-todo"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
  <ul
    className="todos"
  />
</div>
`;
